safe/impact_functions/inundation/tsunami_raster_road/impact_function.py

```diff
@@ -95,8 +99,12 @@ def _raster_to_vector_cells(raster, ranges, output_crs):
 
     # prepare coordinate transform to reprojection
     ct = QgsCoordinateTransform(raster.crs(), output_crs)
-
     rd = 0
+    # Hack because QGIS does not return the
+    # nodata value from the dataset properly in the python API
+    dataset = gdal.Open(raster.source())
+    no_data = dataset.GetRasterBand(1).GetNoDataValue()
+    dataset = None  # close the dataset
```

Comment:
Ok. It's not very pythonic :)
I just read http://gis.stackexchange.com/questions/80366/why-close-a-dataset-in-gdal-python but it's not the coding guide